src/main/java/seedu/address/model/order/predicates/UniqueIdPredicate.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.order.predicates;
+
+import seedu.address.commons.core.index.UniqueId;
+import seedu.address.model.order.Order;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Order}'s {@code UniqueId} matches the Unique Id given.
+ */
+public class UniqueIdPredicate<T extends Order> implements Predicate<T> {
+    private final UniqueId uniqueId;
+
+    public UniqueIdPredicate(UniqueId uniqueId) {
+        this.uniqueId = uniqueId;
+    }
+
+    @Override
+    public boolean test(T order) {
+        String uniqueId = order.getId().getIdToString();
+        boolean isOrder = order.getId().getIdToString().equals(uniqueId);
+        return isOrder;
+    }
```

Comment:
I am guessing the issue is here. 
Perhaps can try:
`String thisId = unique.getIdToString();`
`boolean isOrder = order.getId().getIdToString().equals(thisId);`